UnityProject/Assets/Scripts/Items/Implants/Organs/Stomach.cs

```diff
@@ -1,89 +1,107 @@
-﻿using System.Collections.Generic;
-using Chemistry.Components;
+﻿using Chemistry.Components;
+using System;
+using Chemistry;
 using HealthV2;
+using Items.Food;
+using UnityEngine;
+using System.Collections.Generic;
 
 namespace Items.Implants.Organs
 {
-	public class Stomach : BodyPartFunctionality
+	public class Stomach : BodyPartFunctionality, IStomachProcess
 	{
-		public ReagentContainer StomachContents;
+		//General Stomach Class does not contain functions for fat as only organic stomachs will produce it. See FatProducingStomach.cs
 
-		public float DigesterAmountPerSecond = 1;
+		[SerializeField] protected ItemStorage stomachContents;
+			
+		[SerializeField] protected ReagentContainer reagentContainer;
 
-		public List<BodyFat> BodyFats = new List<BodyFat>();
+		[SerializeField] protected List<Reagent> nutrientReagents; //What reagents this stomach treats as food. Nutrient is base, but stomachs like Moths might consume different reagents for food.
 
-		public BodyFat BodyFatToInstantiate;
+		[SerializeField] protected int DigesterAmountPerSecond = 1; //On average takes 18 seconds to deplete one nutrient.
 
-		public bool InitialFatSpawned = false;
+		public bool AddObjectToStomach(Consumable edible)
+		{
+			return Inventory.ServerAdd(edible.gameObject, stomachContents.GetNextFreeIndexedSlot(), ReplacementStrategy.Cancel);
+		}
 
-		public override void ImplantPeriodicUpdate()
+		public float TryAddReagentsToStomach(ReagentMix reagentMix)
 		{
-			base.ImplantPeriodicUpdate();
+			float consumedAmount = Math.Max(reagentMix.Total, reagentContainer.SpareCapacity);
+			
+			reagentContainer.Add(reagentMix.Take(consumedAmount));
+
+			return consumedAmount;
+		}
 
-			//BloodContainer
-			if (StomachContents.ReagentMixTotal > 0)
+		public virtual float ProcessContent() 
+		{
+			foreach(ItemSlot slot in stomachContents.GetItemSlots()) //Gets reagent from food in stomach, and then despawns food once digested.
 			{
-				float ToDigest = DigesterAmountPerSecond * RelatedPart.TotalModified;
-				if (StomachContents.ReagentMixTotal < ToDigest)
+				if (slot.IsEmpty == true) continue;
+
+				ReagentMix chemicalsFromFood = new ReagentMix();
+				if (slot.ItemObject.TryGetComponent<Edible>(out var edible) == false)
 				{
-					ToDigest = StomachContents.ReagentMixTotal;
+					if (slot.ItemObject.TryGetComponent<NotHandEdible>(out var notHandEdible) == false) continue;
+					chemicalsFromFood.Add(notHandEdible.TakeReagentsFromFood(DigesterAmountPerSecond));
```

Comment:
why are we doing it this way, so the player has the object in the stomach and it takes the reagents from the Object?

Reply:
I saw it mentioned in the convo. It allows for more control on how stuff is digested. And if it ends up, some race or creature doesnt it food, like it eats metal, theres no reworking that needs to be done.
Also has some cool gameplay sideeffects, if surgery becomes compantible with non-bodyparts (which it should at some point) it will allows players to cut open stomachs and remove food, maybe someone accidentally eats a hallucingentic burger, and medical takes it out of there stomach.

Reply:
ok cool